### ISSUE TITLE
key_timeout too high and not properly adjustable

### ISSUE THREAD

**Issue by StefanT83** [ros-teleop/teleop_twist_keyboard#29]
Hi,

I would like to be able to teleoperate a robot similar to a joystick: when I release the joystick, the robot would just stop. Instead, when I run this command

> $rosrun teleop_twist_keyboard teleop_twist_keyboard.py _key_timeout:=0.5

the robot continues to move for another 0.5 seconds, somethimes hitting an obstacle or wall. After 0.5 seconds, the robot stops.

I tried to lower the 0.5 value, say to 0.4, but then I get into another issue. When I start the robot from standstill and press the "I" key on the keyboard (meaning requesting to go forward) and check

> $rosrun echo /cmd_vel

it shows that the teleop_twist_keyboard node publishes alternatively the desired value (corresponding to pressing the "I" key on the keyboard) but alternating this with publishing zeros for some mysterious reason. As a consequence the robot starts "shaking" as a consequence of the inputs requesting advancing, then stopping, then advancing etc. In other words the behavior I get is that the robot is not moving forward by just accelerating (as desired).

**How can I lower that 0.5 value to, say 0.05 or even arbitrarily close to zero?**
(My feeling is that the answer has something to do with this info copy-pasted from the readme.md: "It is recommended that you set key_timeout higher than the initial key repeat delay on your system (This delay is 0.5 seconds by default on Ubuntu, but can be adjusted)."

Thank you.

**Comment by trainman419** [comment]
Yes. The trouble here is that the key_timeout expires before your OS generates a repeated key press. 

So, teleop_twist_keyboard gets a keypress, generates a movement command, times out and generates the stop command, all before it gets the repeated key press from the OS to reset the key timer. This causes jerky movement and it's why the recommendation that you've quoted exists:
> It is recommended that you set key_timeout higher than the initial key repeat delay on your system

To fix this, you need to adjust the key repeat rate within your operating system.
Here's docs on how to adjust the key repeat rate in Ubuntu: https://help.ubuntu.com/stable/ubuntu-help/keyboard-repeat-keys.html.en

Hope that helps.
